pkg/backporter/httpcache_test.go

```diff
@@ -0,0 +1,109 @@
+package backporter
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"testing"
+)
+
+//bugzillaCache tests
+
+func TestBugzillaCache(t *testing.T) {
+	testcases := []struct {
+		name      string
+		cachedKey string
+		value     []byte
+		absentKey string
+	}{
+		{
+			name:      "key 1",
+			cachedKey: "key1",
+			value:     []byte("Test string"),
+			absentKey: "keyx",
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			bc := newBugzillaCache()
+			bc.set(tc.cachedKey, tc.value)
+			cachedVal, isCached := bc.get(tc.cachedKey)
+			if !isCached {
+				t.Errorf("key must be present in cache - %s", tc.cachedKey)
+			}
+			if !bytes.Equal(tc.value, cachedVal) {
+				t.Errorf("cached value wrong expected - %s, got - %s", tc.value, cachedVal)
+			}
+			bc.delete(tc.cachedKey)
+			_, isCachedAfterDelete := bc.get(tc.cachedKey)
+			if isCachedAfterDelete {
+				t.Errorf("key present in cache after deletion - %s", tc.cachedKey)
+			}
+			_, isAbsentKeyPresent := bc.get(tc.absentKey)
+			if isAbsentKeyPresent {
+				t.Errorf("invalid key marked ok - %s", tc.absentKey)
+			}
+
+		})
+	}
+}
+
+type transportMock struct {
+	response *http.Response
+	err      error
+}
+
+func (t transportMock) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	return t.response, t.err
+}
+
+func TestRoundTrip(t *testing.T) {
+	testcases := []struct {
+		name  string
+		tmock transportMock
+	}{
+		{
+			name: "Succesful request to mock server",
+			tmock: transportMock{
+				response: &http.Response{
+					Status:     http.StatusText(http.StatusOK),
+					StatusCode: http.StatusOK,
+					Body:       ioutil.NopCloser(bytes.NewBuffer([]byte("some data"))),
+				},
+				err: nil,
```

Comment:
You don't need to explicitly set fields to `nil`, they are by default (if they are pointers)